runtime/src/bank.rs

```diff
@@ -1534,15 +1534,6 @@ impl Bank {
 
         let accounts_data_size_initial = bank.get_total_accounts_stats().unwrap().data_len as u64;
         bank.accounts_data_size_initial = accounts_data_size_initial;
-        if bank
-            .feature_set
-            .is_active(&feature_set::cap_accounts_data_len::id())
-        {
-            let cost_tracker = CostTracker::new_with_account_data_size_limit(Some(
-                MAX_ACCOUNTS_DATA_LEN.saturating_sub(accounts_data_size_initial),
-            ));
-            *bank.write_cost_tracker().unwrap() = cost_tracker;
-        }
```

Comment:
Completely remove the check from `default_with_accounts()` since `finish_init()` is not called here. This will be correctly set when using `Bank::new_with_paths()`, `Bank::new_for_tests()`, Bank::new_for_benches()`, etc.